dbt/scripts/export_qc_town_close_reports.py

```diff
@@ -0,0 +1,228 @@
+    "35": "2",
+    "36": "3",
+    "37": "3",
+    "38": "2",
+    "39": "3",
+    "70": "1",
+    "71": "1",
+    "72": "1",
+    "73": "1",
+    "74": "1",
+    "75": "1",
+    "76": "1",
+    "77": "1",
+}
+
+
+def is_tri(township_code: str, year: int) -> bool:
+    """Helper function to determine if a town in a given year is undergoing
+    triennial reassessment"""
+    try:
+        tri = TOWNSHIP_TO_TRI[township_code]
+    except KeyError:
+        raise ValueError(f"'{township_code}' is not a valid township code")
+    # 2024 is the City reassessment year (tri code 1), so
+    # ((2024 - 2024) % 3) + 1 == 1, and so on for the other two tris
+    return str((year - 2024 % 3) + 1) == tri
+
+
+def parse_args() -> argparse.Namespace:
+    """Helper function to parse arguments to this script"""
+    parser = argparse.ArgumentParser(
+        description=CLI_DESCRIPTION,
+        epilog=CLI_EXAMPLE,
+        # Parse the description and epilog as raw text so that newlines
+        # get preserved
+        formatter_class=argparse.RawTextHelpFormatter,
+    )
+    parser.add_argument(
+        "--target",
+        required=False,
+        default="dev",
+        help="dbt target to use for querying model data, defaults to 'dev'",
+    )
+    parser.add_argument(
+        "--township",
+        required=True,
+        help="Township code to use in filtering query results",
+    )
+    parser.add_argument(
+        "--year",
+        required=False,
+        default=datetime.datetime.now().year,
+        type=int,
+        help="Tax year to use in filtering query results. Defaults to the current year",
+    )
+    parser.add_argument(
+        "--rebuild",
+        action=argparse.BooleanOptionalAction,
+        default=False,
+        help="Rebuild models before exporting",
+    )
+    parser.add_argument(
+        "--print-table-refresh-command",
+        action=argparse.BooleanOptionalAction,
+        default=False,
+        help=(
+            "Print a command that can be run on the server to refresh "
+            "underlying iasWorld tables and exit. Useful if you want to "
+            "refresh table data before running exports"
+        ),
+    )
+    return parser.parse_args()
+
+
+def main():
+    """Main entrypoint for the script"""
+    args = parse_args()
+    tag = "tag:qc_report_town_close"
+    tag_suffix = "tri" if is_tri(args.township, args.year) else "non_tri"
+
+    if args.print_table_refresh_command:
+        # Use `dbt list` on parents to calculate update command for
+        # iasworld sources that are implicated by this call
+        dbt_list_args = [
+            "--quiet",
+            "list",
+            "--target",
+            args.target,
+            "--resource-types",
+            "source",
+            "--output",
+            "json",
+            "--output-keys",
+            "name",
+            "source_name",
+            "--select",
+            f"+{tag}",
+            f"+{tag}_{tag_suffix}",
```

Comment:
**question (non-blocking):** Do we need to worry about the `indirect_selection: cautious` that the selector used?

Reply:
Great question! Per [the docs on indirect selection](https://docs.getdbt.com/reference/global-configs/indirect-selection), we shouldn't need it because it only applies to the `build` and `run` commands:

> Use the `--indirect_selection` flag to `dbt test` or `dbt build` to configure which tests to run for the nodes you specify.